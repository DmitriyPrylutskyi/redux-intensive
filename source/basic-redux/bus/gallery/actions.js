// Types
import {
    SHOW_NEXT_IMAGE,
    SHOW_PREVIOUS_IMAGE,
    SHOW_SELECTED_IMAGE,
} from './types';

export const showNextImage = () => {
    return {
        type: SHOW_NEXT_IMAGE,
    };
};

export const showPreviousImage = () => {
    return {
        type: SHOW_PREVIOUS_IMAGE,
    };
};

export const showSelectedImage = (photoIndex) => {
    return {
        type:    SHOW_SELECTED_IMAGE,
        payload: photoIndex,
    };
};
