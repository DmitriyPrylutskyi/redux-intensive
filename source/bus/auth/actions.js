// Types
import { types } from './types';

// Async
export const signupAsync = (userData) => {
    return {
        type:    types.SIGNUP_ASYNC,
        payload: userData,
    };
};

export const loginAsync = (credentials) => {
    return {
        type:    types.LOGIN_ASYNC,
        payload: credentials,
    };
};

export const authenticateAsync = () => {
    return {
        type: types.AUTHENTICATE_ASYNC,
    };
};

export const initializeAsync = () => {
    return {
        type: types.INITIALIZE_ASYNC,
    };
};

export const logoutAsync = () => {
    return {
        type: types.LOGOUT_ASYNC,
    };
};

// Sync
export const authenticate = () => {
    return {
        type: types.AUTHENTICATE,
    };
};

export const initialize = () => {
    return {
        type: types.INITIALIZE,
    };
};

export const logout = () => {
    return {
        type: types.LOGOUT,
    };
};
