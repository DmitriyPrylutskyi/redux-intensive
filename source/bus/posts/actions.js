// Types
import { types } from './types';

// Async
export const fetchPostsAsync = () => {
    return {
        type: types.FETCH_POSTS_ASYNC,
    };
};
export const likePostAsync = (postId) => {
    return {
        type:    types.LIKE_POST_ASYNC,
        payload: postId,
    };
};

export const unlikePostAsync = (postId) => {
    return {
        type:    types.UNLIKE_POST_ASYNC,
        payload: postId,
    };
};

// Sync
export const fillPosts = (posts) => {
    return {
        type:    types.FILL_POSTS,
        payload: posts,
    };
};

export const likePost = (likedPostData) => {
    return {
        type:    types.LIKE_POST,
        payload: likedPostData,
    };
};

export const unlikePost = (unlikedPostData) => {
    return {
        type:    types.UNLIKE_POST,
        payload: unlikedPostData,
    };
};
