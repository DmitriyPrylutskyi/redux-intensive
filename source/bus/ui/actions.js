// Types
import { types } from './types';

// Sync
export const startFetching = () => {
    return {
        type: types.START_FETCHING,
    };
};

export const stopFetching = () => {
    return {
        type: types.STOP_FETCHING,
    };
};

export const setOnlineState = () => {
    return {
        type: types.SET_ONLINE_STATE,
    };
};

export const setOfflineState = () => {
    return {
        type: types.SET_OFFLINE_STATE,
    };
};
