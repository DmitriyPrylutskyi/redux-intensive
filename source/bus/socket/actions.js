// Instruments
import { socket } from '../../init/socket';
import { setOnlineState, setOfflineState } from '../ui/actions';
import { createPost, removePost, likePost, unlikePost } from '../posts/actions';
import { Map } from 'immutable';

// Api
import { api } from '../../API';

export const listenConnection = () => {
    return (dispatch) => {
        socket.on('connect', () => {
            dispatch(setOnlineState());
        });

        socket.on('disconnect', () => {
            dispatch(setOfflineState());
        });
    };
};

export const listenPosts = () => {
    return (dispatch, getState) => {
        socket.on('like', async (event) => {
            const response = JSON.parse(event);
            if (response.meta.action === 'dislike') {
                dispatch(unlikePost(response.data));
            } else {
                const { firstName, lastName } = await api.users.getById.call(
                    api,
                    response.data.userId,
                );

                const liker = Map({
                    firstName,
                    lastName,
                    id: response.data.userId,
                });

                dispatch(
                    likePost({
                        postId: response.data.postId,
                        liker,
                    }),
                );
            }
        });

        socket.on('create', (event) => {
            const response = JSON.parse(event);
            dispatch(createPost(response.data));
        });

        socket.on('remove', (event) => {
            const response = JSON.parse(event);
            dispatch(removePost(response.data));
        });
    };
};
