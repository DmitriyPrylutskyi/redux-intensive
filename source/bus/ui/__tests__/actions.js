// Actions
import * as actions from '../actions';

describe('ui actions:', () => {
    test('startFetching', () => {
        expect(actions.startFetching()).toMatchSnapshot();
    });

    test('stopFetching', () => {
        expect(actions.stopFetching()).toMatchInlineSnapshot(`
Object {
  "type": "STOP_FETCHING",
}
`);
    });
    test('setOnlineState', () => {
        expect(actions.setOnlineState()).toMatchInlineSnapshot(`
Object {
  "type": "SET_ONLINE_STATE",
}
`);
    });
    test('setOfflineState', () => {
        expect(actions.setOfflineState()).toMatchInlineSnapshot(`
Object {
  "type": "SET_OFFLINE_STATE",
}
`);
    });
});
