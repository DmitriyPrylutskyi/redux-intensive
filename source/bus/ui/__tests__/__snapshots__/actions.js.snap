// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: startFetching 1`] = `
Object {
  "type": "START_FETCHING",
}
`;
